src/main/java/com/basho/riak/client/api/commands/kv/FullBucketRead.java

```diff
@@ -0,0 +1,321 @@
+            }
+        }
+        private final BuilderFullBucketRead2i builder2i;
+
+        /**
+         * Construct a Builder for a FullBucketRead with a cover context.
+         *
+         * <p>
+         * Note that this command mustn't be used without coverage context for querying buckets
+         * that contain a big amount of data.
+         * <p>
+         */
+        public Builder(Namespace namespace)
+        {
+            builder2i = new BuilderFullBucketRead2i(namespace);
+        }
+
+        /**
+         * Construct a Builder for a FullBucketRead with a cover context.
+         * <p>
+         * Note that in case when query executed on the Riak node other than the one specified by the coverage context,
+         * nothing will be returned.
+         * <p>
+         */
+        public Builder(Namespace namespace, byte[] coverageContext)
+        {
+            builder2i = new BuilderFullBucketRead2i(namespace, coverageContext);
+        }
+
+        private BuilderFullBucketRead2i get2iBuilder()
+        {
+            return builder2i;
+        }
+
+        /**
+         * Return the object (including any siblings).
+         * @param returnBody true to return the object.
+         * @return a reference to this object.
+         */
+        public Builder withReturnBody(boolean returnBody)
+        {
+            builder2i.withReturnBody(returnBody);
+            return self();
+        }
+
+        /**
+         * Set the continuation for this query.
+         * <p>
+         * The continuation is returned by a previous paginated query.
+         * </p>
+         * @param continuation
+         * @return a reference to this object.
+         */
+        public Builder withContinuation(BinaryValue continuation)
+        {
+            builder2i.withContinuation(continuation);
+            return self();
+        }
+
+        /**
+         * Set the maximum number of results returned by the query.
+         * @param maxResults the number of results.
+         * @return a reference to this object.
+         */
+        public Builder withMaxResults(Integer maxResults)
+        {
+            builder2i.withMaxResults(maxResults);
+            return self();
+        }
+
+        /**
+         * Set whether to sort the results of a non-paginated 2i query.
+         * <p>
+         * Setting this to true will sort the results in Riak before returning them.
+         * </p>
+         * <p>
+         * Note that this is not recommended for queries that could return a large
+         * result set; the overhead in Riak is substantial.
+         * </p>
+         *
+         * @param orderByKey true to sort the results, false to return as-is.
+         * @return a reference to this object.
+         */
+        public Builder withPaginationSort(boolean orderByKey)
+        {
+            builder2i.withPaginationSort(orderByKey);
+            return self();
+        }
+
+        protected Builder self()
```

Comment:
It was added to allow descendants to return coresponding self to prevent narrowing API by returning the parent's self. It seems that it is not a case, since we have no descendants at all, so shall I remove it?